### PR TITLE
Added votes to header + added secp256k1 eth sigs

### PR DESCRIPTION
**Pull request by vaibhavchellani**

### CODE REVIEW

**Comment by vaibhavchellani** [comment]
@jdkanani Please review latest-commit

File: types/protobuf.go [hunk 1]
```diff
@@ -39,6 +41,12 @@ var TM2PB = tm2pb{}
 type tm2pb struct{}
 
 func (tm2pb) Header(header *Header) abci.Header {
+	reqBodyBytes := new(bytes.Buffer)
+	//TODO check if this works as supposed to when multiple validators
+	json.NewEncoder(reqBodyBytes).Encode(header.Votes)
+	reqBodyBytes.Bytes()
+
+	fmt.Printf("inside Protobuf.go %v", reqBodyBytes.String())
```
Comment (jdkanani) [inline types/protobuf.go]: Remove unwanted prints

File: types/protobuf.go [hunk 2]
```diff
@@ -39,6 +41,12 @@ var TM2PB = tm2pb{}
 type tm2pb struct{}
 
 func (tm2pb) Header(header *Header) abci.Header {
+	reqBodyBytes := new(bytes.Buffer)
+	//TODO check if this works as supposed to when multiple validators
```
Comment (jdkanani) [inline types/protobuf.go:45]: Please check with multiple validators.

File: types/canonical.go [hunk 3]
```diff
@@ -41,6 +42,17 @@ type CanonicalVote struct {
 	ChainID   string
 }
 
+// [peppermint] create RLP vote to decode in contract
+type CanonicalRLPVote struct {
+	ChainID string
+	// TODO Make sure this works in contract
```
Comment (jdkanani) [inline types/canonical.go]: Please check and make sure before we go ahead

**Review by jdkanani — CHANGES_REQUESTED** [review]

**Comment by jdkanani** [comment]
@vaibhavchellani Any timeline for this?

**Comment by vaibhavchellani** [comment]
Should be done in the next couple of days, undergoing testing with heimdall.

**Comment by jdkanani** [comment]
Also test with root contracts and checkpoints.

**Pull request closed without merging**